src/build/_util.py

```diff
@@ -71,8 +71,15 @@ def parse_wheel_filename(filename: str) -> re.Match[str] | None:
 
 else:
     # Per https://peps.python.org/pep-0706/, the "data" filter will become
-    # the default in Python 3.14.
-    if sys.version_info < (3, 14) and hasattr(tarfile, 'data_filter'):
+    # the default in Python 3.14. The first series of releases with the filter
+    # had a broken filter that could not process symlinks correctly.
+    if (
+        (3, 8, 19) <= sys.version_info < (3, 9)
```

Comment:
```suggestion
        (3, 8, 18) <= sys.version_info < (3, 9)
```

https://docs.python.org/release/3.8.18/whatsnew/changelog.html#library